ban/commands/ignsna.py

```diff
@@ -0,0 +1,48 @@
+import os
+from ban.commands import command, report
+from ban.core.models import (Municipality, PostCode)
+from .helpers import session, batch, nodiff
+
+__namespace__ = 'import'
+
+
+@command
+@nodiff
+def ignsna(path, **kwargs):
+    """Import from IGN/Laposte BDUNI
+    :param path: directory location of the files hsp7aaaa.ai, hsv7aaaa.ai and hsw4aaaa.ai"""
+
+    municipality_zipcode_file = os.path.join(path, 'hsp7aaaa.ai')
+    if municipality_zipcode_file is not None:
+        lines = open(municipality_zipcode_file).readlines()
```

Comment:
You are consuming twice the file without resetting it, that's why.
If the file is not too big, you can keep using `readlines`.
Otherwise, compute the length with the `count` trick you were using before (maybe move it to helpers), and then call `f.seek(0)` to reset the cursor.